oneapp-server/dao/ApplicationProgramInfoDao.js

```diff
@@ -0,0 +1,38 @@
+const { SQLDataSource } = require('datasource-sql');
+const oracledb = require('oracledb');
+const { OneAppError } = require('../utils/OneAppError');
+
+class ApplicationProgramInfoDao extends SQLDataSource {
+  async getProgramInfo(APPLICATION_ID, LANGUAGE) {
+    const con = await this.knex.client.pool.acquire().promise;
+    return this.knex.client.transaction(async (tx) => {
+      const bindVars = {
+        app_id: { dir: oracledb.BIND_IN, val: APPLICATION_ID },
+        lang: { dir: oracledb.BIND_IN, val: LANGUAGE },
+        info: { dir: oracledb.BIND_OUT },
+      };
+      return tx.raw('begin OA_PKG_APP.SP_SELECT_APPLICATION_PROGRAMS(:app_id, :lang, :info); end;', bindVars);
```

Comment:
Response should return the first result in the array.

Reply:
changed the response value to return the program info object

Reply:
Interesting, so specifying a type on the `BIND_OUT` stops the Oracle driver from returning an array?